### PR TITLE
Jonmv/cross instance region application endpoints

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/ApplicationPackageValidator.java [hunk 1]
```diff
@@ -90,22 +93,40 @@ private void validateSteps(DeploymentSpec deploymentSpec) {
...
-                    throw new IllegalArgumentException("Endpoint '" + endpoint.endpointId() + "' in " + instance +
-                                                       " cannot contain regions in different clouds: " +
+            validateEndpointRegions(instance.endpoints(), instance);
+        }
+        validateEndpointRegions(deploymentSpec.endpoints(), null);
+    }
+
+    private void validateEndpointRegions(List<Endpoint> endpoints, DeploymentInstanceSpec instance) {
+        for (var endpoint : endpoints) {
+            RegionName[] regions = new HashSet<>(endpoint.regions()).toArray(RegionName[]::new);
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/ApplicationPackageValidator.java:111]: Why `toArray`?

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java [hunk 2]
```diff
@@ -636,63 +638,107 @@ void testDnsUpdatesForApplicationEndpoint() {
...
+                                                     RecordData.from("weighted/lb-0--tenant1.app1.main--prod.aws-us-east-1a/dns-zone-1/prod.aws-us-east-1a/1")),
+                                          new Record(Record.Type.ALIAS,
+                                                     RecordName.from("c.app1.tenant1.aws-us-east-1b-r.vespa.oath.cloud"),
+                                                     RecordData.from("weighted/lb-0--tenant1.app1.beta--prod.aws-us-east-1b/dns-zone-1/prod.aws-us-east-1b/4")),
+                                          new Record(Record.Type.CNAME,
+                                                     RecordName.from("d.app1.tenant1.us-west-1-r.vespa.oath.cloud"),
+                                                     RecordData.from("vip.prod.us-west-1.")),
+                                          new Record(Record.Type.CNAME,
+                                                     RecordName.from("e.app1.tenant1.us-east-3-r.vespa.oath.cloud"),
+                                                     RecordData.from("vip.prod.us-east-3.")))),
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java:729]: 😅

**Review by mpolden — COMMENTED** [review]
Missing change to schema? (`deployment.rnc`)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/ApplicationPackageValidator.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/ApplicationPackageValidator.java:111]: That's the required argument for the `.in(RegionName... regions)`.

**Review by mpolden — APPROVED** [review]

**Pull request merged**